daemon/config/config_test.go

```diff
@@ -387,106 +378,82 @@ func (tf overwriteTransformer) Transformer(typ reflect.Type) func(dst, src refle
 }
 
 func TestValidateConfiguration(t *testing.T) {
+
+	config0, _ := New()
+	config0.CommonConfig.Labels = []string{"one=two"}
+	config1, _ := New()
+	config1.CommonConfig.DNSConfig.DNSSearch = []string{"a.b.c"}
+	config2, _ := New()
+	config2.CommonConfig.BridgeConfig.DefaultBridgeConfig.MTU = 1234
+	config3, _ := New()
+	config3.CommonConfig.MaxConcurrentDownloads = 4
+	config4, _ := New()
+	config4.CommonConfig.MaxConcurrentUploads = 4
+	config5, _ := New()
+	config5.CommonConfig.MaxDownloadAttempts = 4
+	config6, _ := New()
+	config6.CommonConfig.NodeGenericResources = []string{"foo=bar", "foo=baz"}
+	config7, _ := New()
+	config7.CommonConfig.NodeGenericResources = []string{"foo=1"}
+	config8, _ := New()
+	config8.CommonConfig.Hosts = []string{"tcp://127.0.0.1:2375"}
+	config9, _ := New()
+	config9.CommonConfig.LogLevel = "warn"
```

Comment:
Why is this change necessary? Like the other modified table-driven test, this one also uses mergo to merge the config partial in the table on top of defaults.

Reply:
I was asked to modify the test cases to use the New() constructor

Reply:
That's my point: this test _already_ uses the New() constructor! https://github.com/moby/moby/blob/bed37b6152327ae67f37cebf2690b7d746b99fb6/daemon/config/config_test.go#L494-L497

Reply:
That is merging the configs.

Reply:
So? Why is that significant?